americangut/diversity_analysis.py

```diff
@@ -0,0 +1,1729 @@
+        Default is str. The format for xtick label text if using the
+        default tick values as the tick labels.
+    ytick_format : class, optional
+        Default is str. The format for xtick label text if using the
+        default tick values as the tick labels.
+    """
+    kwds = {'counts': None,
+            'n_xs': None,
+            'n_y': None,
+            'n_size': 11,
+            'p_value': None,
+            'p_x': None,
+            'p_y': None,
+            'p_size': 12,
+            'show_frame': True,
+            'show_xticks': True,
+            'show_yticks': True,
+            'show_xgrid': False,
+            'show_ygrid': False,
+            'title': '',
+            'title_size': 18,
+            'xlim': None,
+            'xlabel': '',
+            'xticks': None,
+            'xticklabels': None,
+            'xtick_format': '%s',
+            'xfont_align': 'center',
+            'xfont_angle': 0,
+            'xtick_size': 12,
+            'xlabel_size': 15,
+            'ylim': None,
+            'yticks': None,
+            'yticklabels': "text",
+            'ytick_format': '%s',
+            'ylabel': '',
+            'yfont_angle': 0,
+            'yfont_align': 'right',
+            'ylabel_size': 15,
+            'ytick_size': 12}
+
+    for key, val in kwargs.iteritems():
+        if key in kwds:
+            kwds[key] = val
+        else:
+            raise ValueError('%s is not an input for axis formatting.' % key)
+
+    # Sets axis ticks
+    if kwds['xticks'] is not None:
+        ax.set_xticks(kwds['xticks'])
+
+    if kwds['yticks'] is not None:
+        ax.set_yticks(kwds['yticks'])
+
+    # Checks the max and min values values
+    if kwds['xlim'] is not None and not len(kwds['xlim']) == 2:
+        raise ValueError('xlim must specify a xmin and xmax value.')
+
+    if kwds['ylim'] is not None and not len(kwds['ylim']) == 2:
+        raise ValueError('xlim must specify a ymin and ymax value.')
+
+    # Sets axis limits
+    if kwds['xlim'] is not None:
+        ax.set_xlim(kwds['xlim'])
+    if kwds['ylim'] is not None:
+        ax.set_ylim(kwds['ylim'])
+
+    # Formats the tick labels
+    if kwds['xticklabels'] is None:
+        xtls = ['']*len(ax.get_xticks())
+    elif (isinstance(kwds['xticklabels'], str) and
+          kwds['xticklabels'].lower() == 'text'):
+        xtls = [kwds['xtick_format'] % t for t in ax.get_xticks()]
+    elif not isinstance(kwds['xticklabels'], (list, tuple, np.ndarray)):
+        raise TypeError('xticklabels must be None, "text" or an iterable.')
+    elif not len(kwds['xticklabels']) == len(ax.get_xticks()):
+        raise ValueError('There must be a label for each xtick.')
+    else:
+        xtls = kwds['xticklabels']
+
+    ax.set_xticklabels(xtls,
+                       ha=kwds['xfont_align'],
+                       rotation=kwds['xfont_angle'],
+                       size=kwds['xtick_size'])
+
+    if kwds['yticklabels'] is None:
```

Comment:
conditional block appears to be essentially the same as the one above, please decompose (could just do a local function or something)